example/fish/chatRoomFish.ts

```diff
@@ -54,52 +34,24 @@ export type MessageEvent = {
 }
 export type Event = MessageEvent
 
-export const onEvent: OnEvent<State, Event> = (state: State, event: Envelope<Event>) => {
-  const { payload } = event
-  switch (payload.type) {
+export const onEvent: Reduce<State, Event> = (state: State, event: Event) => {
+  switch (event.type) {
     case EventType.message: {
-      const newMessage = `${payload.sender}: ${payload.message}`
-      return [newMessage, ...state]
+      const newMessage = `${event.sender}: ${event.message}`
+      state.unshift(newMessage)
     }
   }
   return state
 }
 
-/**
- * Fish Commands
- */
-export enum CommandType {
-  postMessage = 'postMessage'
-}
-export type PostMessageCommand = {
-  type: CommandType.postMessage
-  sender: string
-  message: string
+const tags = {
+  channel: Tag<Event>('channel')
 }
-export type Command = PostMessageCommand
-
-export const onCommand: OnCommand<State, Command, Event> = (_state: State, command: Command) => {
-  switch (command.type) {
-    case CommandType.postMessage: {
-      return [
-        {
-          type: EventType.message,
-          message: command.message,
-          sender: command.sender
-        }
-      ]
-    }
-  }
-  return []
-}
-
-/*
- * Fish Definition
- */
-export const ChatRoomFish = FishType.of<State, Command, Event, PublicState>({
-  semantics: Semantics.of('com.example.chatRoom'),
-  initialState,
+const forChannel = (channel: string): Fish<State, Event> => ({
+  fishId: FishId.of('com.example.chatRoom', channel, 0),
+  initialState: [],
   onEvent,
-  onCommand,
-  onStateChange: OnStateChange.publishPrivateState()
+  where: tags.channel.withId(channel)
 })
+
+export const ChatRoomFish = { forChannel, tags }
```

Comment:
Personally, I struggle a bit to give this the same name as the old Fish. But probably its a matter of time.

Reply:
I can feel you. Maybe
```suggestion
export const ChatRoom = { fishForChannel, tags }
```
would be better?

Reply:
so simple, but much better :-)